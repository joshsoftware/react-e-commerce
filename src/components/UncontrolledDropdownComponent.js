import React from 'react';
import DropdownToggleComponent from './DropdownToggleComponent';
import DropdownMenuComponent from './DropdownMenuComponent';
import { UncontrolledDropdown } from 'reactstrap';
import { useSelector } from 'react-redux';

const UncontrolledDropdownComponent = () => {
  const { profile } = useSelector((state) => state.userprofileReducer);
  const profileImage =
    profile.profile_image !== ''
      ? `${process.env.REACT_APP_SERVER_URL}${profile.profile_image}`
      : 'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png';
  return (
    <UncontrolledDropdown nav inNavbar>
      <DropdownToggleComponent src={profileImage} className="rounded" alt="img" />
      <DropdownMenuComponent />
    </UncontrolledDropdown>
  );
};

export default UncontrolledDropdownComponent;
