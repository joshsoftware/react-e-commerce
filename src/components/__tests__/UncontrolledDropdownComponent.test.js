import UncontrolledDropdownComponent from '../UncontrolledDropdownComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('UncontrolledDropdown Component', () => {
  it('Must return UncontrolledDropdown Component', () => {
    const { asFragment } = render(<UncontrolledDropdownComponent />);

    expect(asFragment()).toMatchSnapshot();
  });
});
