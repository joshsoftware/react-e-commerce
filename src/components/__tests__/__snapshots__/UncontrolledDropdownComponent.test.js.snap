// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UncontrolledDropdown Component Must return UncontrolledDropdown Component 1`] = `
<DocumentFragment>
  <li
    class="dropdown nav-item"
  >
    <a
      aria-expanded="false"
      aria-haspopup="true"
      class="dropdown-toggle nav-link"
      href="#"
    >
      <img
        alt="img"
        class="rounded"
        src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
        style="width: 50px;"
      />
    </a>
    <div
      aria-hidden="true"
      class="dropdown-menu dropdown-menu-right"
      role="menu"
      tabindex="-1"
    >
      <button
        class="dropdown-item"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        Profile
      </button>
      <div
        class="dropdown-divider"
        tabindex="-1"
      />
      <button
        class="dropdown-item"
        role="menuitem"
        tabindex="0"
        type="button"
      >
        Logout
      </button>
      <div
        class="dropdown-divider"
        tabindex="-1"
      />
    </div>
  </li>
</DocumentFragment>
`;
